tests/models/hf/test_gpt_bigcode_hf.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+from transformers import (
+    PreTrainedModel,
+    GPTBigCodeForCausalLM,
+    GPTBigCodeConfig,
+    PretrainedConfig,
+    PreTrainedTokenizer,
+)
+import torch
+
+from fms.models.gpt_bigcode import GPTBigCode
+from fms.models.hf.gpt_bigcode.configuration_gpt_bigcode_hf import (
+    HFAdaptedGPTBigCodeConfig,
+)
+from fms.models.hf.gpt_bigcode.modeling_gpt_bigcode_hf import (
+    HFAdaptedGPTBigCodeForCausalLM,
+)
+from fms.testing._internal.hf.model_test_suite import (
+    HFConfigFixtureMixin,
+    HFModelFixtureMixin,
+    HFConfigTestSuite,
+    HFModelEquivalenceTestSuite,
+    HFModelGenerationTestSuite,
+    HFModelCompileTestSuite,
+)
+from fms.testing._internal.model_test_suite import ModelFixtureMixin
+from ..test_gpt_bigcode import GPTBigCodeFixtures
+
+
+class HFAdaptedGPTBigCodeFixtures(
+    ModelFixtureMixin, HFConfigFixtureMixin, HFModelFixtureMixin
+):
+    @pytest.fixture(scope="class", autouse=True)
+    def fms_hf_model(
+        self, model: GPTBigCode, fms_hf_config: PretrainedConfig, **kwargs
+    ):
+        return HFAdaptedGPTBigCodeForCausalLM.from_fms_model(
+            model, **fms_hf_config.to_dict()
+        )
+
+    @pytest.fixture(scope="class", autouse=True)
+    def fms_hf_config(
+        self, tokenizer: PreTrainedTokenizer, model: GPTBigCode, **kwargs
+    ) -> PretrainedConfig:
+        bos_token_id = (
+            tokenizer.bos_token_id
+            if tokenizer.bos_token_id is not None
+            else tokenizer.eos_token_id
+        )
+        return HFAdaptedGPTBigCodeConfig.from_fms_config(
+            model.get_config(),
+            eos_token_id=tokenizer.eos_token_id,
+            bos_token_id=bos_token_id,
+        )
+
+    @pytest.fixture(scope="class", autouse=True)
+    def oss_hf_model(
+        self, fms_hf_model: HFAdaptedGPTBigCodeForCausalLM
+    ) -> PreTrainedModel:
+        hf_config = fms_hf_model.config
+        oss_hf_model = GPTBigCodeForCausalLM(
```

Comment:
this could also be made into a function in `fms/models/hf/gpt_bigcode/__init__.py` to turn an FMS model to an HF one

Reply:
I added a method `convert_to_hf` which will convert an HF-Adapted FMS GPTBigCode model to an HF model